### ISSUE TITLE
Support for credential_source in profile file

### ISSUE THREAD

**Issue by rce** [aws/aws-sdk-java-v2#1169]
## Expected Behavior

I'd expect the same support for `credential_source` as the CLI.

I can work on this if the change is wanted.

## Current Behavior

Currently profiles with only `iam_role` and `credential_source` properties are ignored.

## Possible Solution

Change profile parsing to construct proper credential providers when a profile with `credential_source` is found.

## Context

I have EC2 instance with multiple profiles in config file for cross-account role access and the instance role has permissions to assume the roles. This can be worked around but I think expecting same behaviour as the CLI is reasonable.

## Your Environment

* AWS Java SDK version used: 2.5.4
* JDK version used: 1.8.0_192
* Operating System and version: macOS 10.14.3

**Comment by zoewangg** [comment]
Thank you for providing the PR! We will take a look shortly.

**Comment by jhigueras** [comment]
> ## Context
> I have EC2 instance with multiple profiles in config file for cross-account role access and the instance role has permissions to assume the roles. **This can be worked around but I think expecting same behaviour as the CLI is reasonable.**

Could you share the workaround? I'm in a similar situation, where i have the default profile with a role_arn to assume a cross account role, and the source_credentials=EcsContainer. In our case, the ecs task role is the one with permissions to assume the target role. With the cli and boto3 it works fine, but with java we're stuck on this.

**Comment by rce** [comment]
@jhigueras We just create the StsAssumeRoleCredentialsProvider manually.

For example in Scala:

```scala
// Instead of just S3Client.create() tell the client explicitly to use credentials provider for the required role
val s3Client = S3Client.builder()
  .credentialsProvider(roleCredentialsProvider("arn:aws:iam::1234567890:role/foo"))
  .build()

def roleCredentialsProvider(roleArn: String): StsAssumeRoleCredentialsProvider = {
  // This would be the credentials provider matching `credential_source` in profile definition.
  // That is ContainerCredentialsProvider.builder().build() for `EcsContainer`,
  // InstanceProfileCredentialsProvider.create() for `Ec2InstanceMetadata` and
  // EnvironmentVariableCredentialsProvider.create() for `Environment`
  val credentialSource = InstanceProfileCredentialsProvider.create()

  val stsClient = StsClient.builder().credentialsProvider(credentialSource).build()
  val request = AssumeRoleRequest.builder()
    .roleArn(roleArn)
    .roleSessionName("bar")
    .build()
  StsAssumeRoleCredentialsProvider.builder()
    .refreshRequest(request)
    .stsClient(stsClient)
    .build()
}
```

**Comment by zoewangg** [comment]
Implemented via #1179 and released in `2.5.30`
